lib/src/main/scala/com/worksap/nlp/uzushio/lib/filters/NoContentDOM.scala

```diff
@@ -1,26 +1,90 @@
 package com.worksap.nlp.uzushio.lib.filters
 
-import com.worksap.nlp.uzushio.lib.cleaning.Paragraph
+import com.worksap.nlp.uzushio.lib.cleaning.{Paragraph, PathSegment}
 import com.worksap.nlp.uzushio.lib.filters.base.ParagraphFilter
 
 class NoContentDOM extends ParagraphFilter {
-  // This names are tag names, but also class names and id names
   final private val filteringDomNames: Seq[String] =
-    Array("header", "footer", "aside", "nav", "left-box", "sidebar")
+    Array("header", "footer", "aside", "nav", "noscript", "form")
+
+  final private val DOMCandidatesForFiliteringClassOrId = Array("div", "p", "ul", "h1")
+
+  final private val filteringFullMatchClassOrIdCandidates: Seq[String] = Array(
```

Comment:
there is a relatively long list of candidates, probably it's better to use sets here